File: 217-contains-duplicate/contains-duplicate.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> result = new HashSet<Integer>();
        for(int i: nums){
            if(result.contains(i)) return true;
            result.add(i);
        }
        return false;
    }
}

File: 217-contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        cache = set()
        for i in nums:
            if i in cache:
                return True
            cache.add(i)
        return False



        # results = set()
        # for i in nums:
        #     if i in results:
        #         return True
        #     results.add(i)
        # return False
        